### PR TITLE
NumPad to enter a number from arbitrary range easily

### PR DESCRIPTION
**Pull request by tsusanka**
To easily enter numbers from an arbitrary range. I've decided to go with the design where as much buttons are present as possible, because then 32 (the max number of shares in SLIP-39) fits in three pages.

peek 2019-02-21 09-45

For a perfect UI maybe a back button should be present when you click some + button? So you can go back

### CODE REVIEW

File: src/trezor/ui/num_pad.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+def digit_area(i):
+    return ui.grid(i + 3)  # skip the first line
+
+
+class NumPad(ui.Widget):
+    def __init__(self, label: str, start: int, end: int):
+        """
+        Generates a numpad with numbers from `start` to `end` excluding.
+        """
+        display.clear()
```
Comment (jpochyla) [inline src/trezor/ui/num_pad.py]: This shouldn't really be needed, `ui.layout` usually takes care of clearing the display.

File: src/trezor/ui/num_pad.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        for btn in self.buttons:
+            if btn.touch(event, pos) == BTN_CLICKED:
+                if "+" in btn.content:
+                    self.page += 1
+                    self._generate_buttons()
+                else:
+                    return btn.content
+
+    def _generate_buttons(self):
+        display.clear()
```
Comment (jpochyla) [inline src/trezor/ui/num_pad.py]: Please add a code comment explaining why we are clearing the screen here.

Comment (tsusanka) [reply src/trezor/ui/num_pad.py]: Hm, because we need to redraw the buttons with other values. I can comment, but it makes sense, doesn't it?

Comment (tsusanka) [reply src/trezor/ui/num_pad.py]: This is what I get if I don't:

emu00000001

The last row should not be there at all.

**Pull request merged**